### PR TITLE
Exempt TypedDict from too-few-public-methods check

### LINKED ISSUE #4180: False positive too-few-public-methods for TypedDict in Python 3.9
This is a follow up to #1927, #2607, and #4072. In Python 3.9, TypedDict was changed from a class to a function with metaclass magic. This change caused worse problems initially (fixed by #4072), but now pylint gives a warning about too few public methods when subclassing TypedDict to create a new type as suggested in the documentation (https://docs.python.org/3/library/typing.html#typing.TypedDict). A work around is to use one of the alternate TypedDict syntax forms in which TypedDict is used as a function instead of a class.

### Steps to reproduce

Given a file `a.py`:
```python
"Demo"
import typing


class Coordinate(typing.TypedDict):
    "This is my type"
    x: float
    y: float
```

### Current behavior

Result of `pylint a.py`:
```
************* Module test
test.py:5:0: R0903: Too few public methods (0/2) (too-few-public-methods)

------------------------------------------------------------------
Your code has been rated at 7.50/10 (previous run: 7.50/10, +0.00)
```

### Expected behavior

No warnings.

### pylint --version output

Result of `pylint --version` output:
```
pylint 2.7.2                                                                                  
astroid 2.5.1                                                                                 
Python 3.9.2
```

### PR DESCRIPTION
**Pull request by cdce8p**
## Steps

- [ ] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [ ] If it's a new feature or an important bug fix, add a What's New entry in `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Description
Don't check `too-few-public-methods` for `typing.TypedDict`.

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :bug: Bug fix  |

## Related Issue
Closes #4180

### CODE REVIEW

File: pylint/checkers/design_analysis.py [hunk 1]
```diff
@@ -95,16 +95,17 @@
...
 def _is_exempt_from_public_methods(node: astroid.ClassDef) -> bool:
     """Check if a class is exempt from too-few-public-methods"""
 
-    # If it's a typing.Namedtuple or an Enum
+    # If it's a typing.Namedtuple, typing.TypedDict or an Enum
     for ancestor in node.ancestors():
         if ancestor.name == "Enum" and ancestor.root().name == "enum":
             return True
-        if ancestor.qname() == TYPING_NAMEDTUPLE:
+        if ancestor.qname() in (TYPING_NAMEDTUPLE, TYPING_TYPEDDICT):
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/design_analysis.py:108]: Shouldn't we create a allow list for class that can have a few public methods in configuration ? The entirety of django models and form classes would fit for example. This is outside the scope of this MR of course but this is the refactor that comes to mind seeing this.

**Review by Pierre-Sassoulas — APPROVED** [review]

File: pylint/checkers/design_analysis.py (hunk 1 above)
Comment (cdce8p) [reply pylint/checkers/design_analysis.py:108]: Sounds reasonable

Comment (Pierre-Sassoulas) [reply pylint/checkers/design_analysis.py:108]: Should I open an issue or could you do it in this MR ? I think this it merge-able as it is.

Comment (cdce8p) [reply pylint/checkers/design_analysis.py:108]: I think that should be done separately. The MR is just a bug fix whereas that would be a new feature.

Comment (Pierre-Sassoulas) [reply pylint/checkers/design_analysis.py:108]: Opened #4197 for it.

**Comment by cdce8p** [comment]
@Pierre-Sassoulas Do you what to add the `2.7.3` milestone here, as well?

**Pull request merged**